homeassistant/components/utility_meter/sensor.py

```diff
@@ -420,6 +420,14 @@ def calculate_adjustment(
     @callback
     def async_reading(self, event: Event):
         """Handle the sensor state changes."""
+        if (
+            source_state := self.hass.states.get(self._sensor_source_id)
+        ) is None or source_state.state == STATE_UNAVAILABLE:
+            self._attr_available = False
```

Comment:
Do we need to write state here?

This could probably use a test

Reply:
absolutely right :)

Reply:
I'm not sure how often this is going to get callled back, but probably should only write it if `self._attr_available` is changing to from `True` to `False`

Reply:
well I picked this code snippet from https://github.com/home-assistant/core/blob/dev/homeassistant/components/switch_as_x/entity.py#L59

Reply:
It looks like we short circuit attributes when something is unavailable so we shouldn't see lots of state writes once it goes unavailable until it goes available again.
https://github.com/home-assistant/core/blob/2f4325246b08225d2974dd7ca08245a780f91292/homeassistant/helpers/entity.py#L633

The worse I think you get is another state write if the entity is removed so likely not a problem